packages/components/src/custom-select-control-v2/legacy-component/index.tsx

```diff
@@ -0,0 +1,135 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import * as Ariakit from '@ariakit/react';
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+/**
+ * Internal dependencies
+ */
+import _CustomSelect from '../custom-select';
+import type { LegacyCustomSelectProps } from '../types';
+import { CustomSelectItem } from '..';
+import * as Styled from '../styles';
+import { ContextSystemProvider } from '../../context';
+
+function _LegacyCustomSelect( props: LegacyCustomSelectProps ) {
+	const {
+		__experimentalShowSelectedHint,
+		__next40pxDefaultSize = false,
+		options,
+		onChange,
+		size = 'default',
+		value: valueProp,
+		...restProps
+	} = props;
+
+	// Forward props + store from v2 implementation
+	const store = Ariakit.useSelectStore( {
+		async setValue( value ) {
+			if ( ! onChange ) return;
+
+			// Executes the logic in a microtask after the popup is closed.
+			// This is simply to ensure the isOpen state matches that in Downshift.
+			await Promise.resolve();
+			const state = store.getState();
+
+			const changeObject = {
+				highlightedIndex: state.renderedItems.findIndex(
+					( item ) => item.value === value
+				),
+				inputValue: '',
+				isOpen: state.open,
+				selectedItem: {
+					// Value will always be a string for the legacy component
+					name: value as string,
+					key: value as string,
```

Comment:
It looks like the Ariakit types can infer from the `defaultValue` type we've passed that `value` will also be `string`!

```suggestion
					name: value,
					key: value,
```

Reply:
Amazing! That wasn't always the case 😅 I should track down why it changed